cylc/flow/cycling/__init__.py

```diff
@@ -41,34 +41,34 @@ def parse_exclusion(expr):
         return remainder.strip(), exclusions
 
 
-class PointBase(object, metaclass=ABCMeta):
+class PointBase(metaclass=ABCMeta):
 
     """The abstract base class for single points in a cycler sequence.
 
     Points should be based around a string value.
 
     Subclasses should provide values for TYPE and TYPE_SORT_KEY.
-    They should also provide self.cmp_, self.sub, self.add, and
+    They should also provide self._cmp, self.sub, self.add, and
     self.eq methods which should behave as __cmp__, __sub__,
-    etc standard comparison methods. Note: "cmp_" not "cmp".
+    etc standard comparison methods. Note: "_cmp" not "cmp".
 
     Subclasses may also provide an overridden self.standardise
     method to reprocess their value into a standard form.
 
     """
 
-    _TYPE = None
-    _TYPE_SORT_KEY = None
-
     __slots__ = ('value')
 
+    @property
     @abstractmethod
-    def TYPE(self):
-        return self._TYPE
+    def TYPE(self) -> str:
+        """Cycling type of this point."""
```

Comment:
This is how you do an abstract property in Python 3.3+
https://stackoverflow.com/a/48710068/3217306